src/node_resource_limits.cc

```diff
@@ -0,0 +1,72 @@
+#include "node_resource_limits.h"
+
+#include <fstream>
+
+#ifdef __POSIX__
+#include <sys/stat.h>
+#endif  // __POSIX__
+
+namespace node {
+namespace resource_limits {
+
+#ifdef __POSIX__
+bool CgroupsResourceLimits::CgroupsAvailable(std::string cgroups_directory) {
+  struct stat stat_buf;
+  int err = stat(cgroups_directory.c_str(), &stat_buf);
+  if (err) {
+    return false;
+  } else {
+    return !!(stat_buf.st_mode & S_IFDIR);
+  }
+}
+
+bool CgroupsResourceLimits::MemParse(uint64_t* output, std::string input) {
+  int64_t value = std::atoll(input.c_str());
+  if (value <= 0 && input != "0") {  // -1 is a valid value
+    return false;
+  }
+  char suffix = input.at(input.length() - 1);
+  uint64_t multiplier = 1;
+  switch (suffix) {
+    case 'g':
+    case 'G':
+      multiplier <<= 10;
+    case 'm':
+    case 'M':
+      multiplier <<= 10;
+    case 'k':
+    case 'K':
+      multiplier <<= 10;
+  }
+  *output = static_cast<uint64_t>(value) * multiplier;
+  return true;
+}
+
+bool CgroupsResourceLimits::GetParamAsBytes(
+    uint64_t* output, std::string controller, std::string param) {
+  std::string file_contents;
+  std::ifstream file;
```

Comment:
This isn't exactly wrong (`std::ifstream` is used in more places) but it leaks a file descriptor when another thread forks when the file is open.

`uv_fs_open()` takes care of marking the file descriptor `O_CLOEXEC`, maybe you can use that?

There's a `WriteFileSync()` in src/util.cc that's a wrapper around `uv_fs_open()`, perhaps you can add a `ReadFileSync()` counterpart?

You can steal the loop logic from `ReadFile()` in src/module_wrap.cc; please refactor that function if you do.